keyboards/ergodone/keymaps/erovia/keymap.c

```diff
@@ -0,0 +1,233 @@
+ *                                 |      |      |      |       |      |      |      |
+ *                                 `--------------------'       `--------------------'
+ */
+// Functions
+[FN] = LAYOUT_ergodox(
+       // left hand
+       VRSN,     KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F11,
+       KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_MPRV,  KC_MPLY,  KC_MNXT,  KC_TRNS,
+       KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_VOLD,  KC_VOLU,
+       KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+       KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+                                                         KC_TRNS,  KC_TRNS,
+                                                                   KC_TRNS,
+                                               KC_TRNS,  KC_TRNS,  KC_TRNS,
+       // right hand
+       KC_F12,   KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_TRNS,
+       KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+                 KC_TRNS,  QWRTY,    CLMK,     KC_TRNS,  KC_TRNS,  KC_TRNS,
+       KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+                           KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,
+       KC_TRNS,  KC_TRNS,
+       KC_TRNS,
+       KC_TRNS,  KC_TRNS,  KC_TRNS
+),
+};
+
+const uint16_t PROGMEM fn_actions[] = {
+    [1] = ACTION_LAYER_TAP_TOGGLE(FN)                // FN1 - Momentary Layer 1
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+      case VRSN:
+        if (record->event.pressed) {
+          SEND_STRING (QMK_KEYBOARD "/" QMK_KEYMAP " @ " QMK_VERSION);
+        }
+        return false;
+        break;
+      case QWRTY:
+        if (record->event.pressed) {
+          set_single_persistent_default_layer(QWERTY);
+        }
+        return false;
+        break;
+      case CLMK:
+        if (record->event.pressed) {
+          set_single_persistent_default_layer(COLEMAK);
+        }
+        return false;
+        break;
+      case KC_CAPS:
+        if (record->event.pressed) {
+          // Turn LED1 On/Off for Caps Lock
+          if (CAPS_LED) {
+            ergodox_right_led_1_off();
+            CAPS_LED = false;
+          } else {
+            ergodox_right_led_1_on();
+            CAPS_LED = true;
+          }
+        }
+        return true;
+        break;
+    }
+    return true;
+};
+
+// Set LED according to the default layer
+void default_layer_led_set(void) {
+    switch (biton32(default_layer_state)) {
+      case COLEMAK:
+        // LED2 for COLEMAK
+        ergodox_right_led_2_on();
+        break;
+      case QWERTY:
+        // LED3 for QWERTY
+        ergodox_right_led_3_on();
+        break;
+    };
+};
+
+// Runs just one time when the keyboard initializes.
+void matrix_init_user(void) {
+    // This function takes care about reading the default layer from EEPROM
+    // and setting the default_layer_state variable.
+    bootmagic();
```

Comment:
This isn't necessary.  This should happen regardless if bootmagic is enabled or not, as it's part of the keyboard init.

Reply:
Ah, I see why you're going that now. 

A better solution would be to use `eeconfig_read_default_layer()`, actually.  This will read it directly from the eeprom, so that it will be the correct values, without having to call a bunch of additional code. 

I actually do this personally, because of the issues you've mentioned: 
https://github.com/qmk/qmk_firmware/blob/master/users/drashna/rgb_stuff.c#L266

Reply:
If you don't want to change this, that's fine. 

But let me know if/when this is ready (but please do add the `#pragma once` to the top of the config.h file, so that it does have an include guard.